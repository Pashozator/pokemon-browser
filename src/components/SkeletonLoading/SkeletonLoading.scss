$main: #e0e1e4;

.skeleton-loading {
  width: 100%;
  height: 20px;
  background: #e2e2e2;
  background: linear-gradient(-45deg, $main, lighten($main, 5), $main);
  background-size: 500% 200%;
  animation: gradientKeyframes 1s ease infinite;
}

@keyframes gradientKeyframes {
  0% {
    background-position: 0 50%;
  }

  50% {
    background-position: 100% 50%;
  }

  100% {
    background-position: 0 50%;
  }
}
