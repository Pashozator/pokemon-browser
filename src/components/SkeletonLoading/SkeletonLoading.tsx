import React from 'react';
import './SkeletonLoading.scss';

interface SkeletonLoadingProps {
	style?: any;
}

const SkeletonLoading: React.FC<SkeletonLoadingProps> = ({ style }) => {
	return (
		<div className="skeleton-loading" style={style}/>
	);
};

export default SkeletonLoading;
