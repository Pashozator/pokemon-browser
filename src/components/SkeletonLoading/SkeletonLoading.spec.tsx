import React from 'react'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SkeletonLoading from './SkeletonLoading';

describe('SkeletonLoading component', () => {
	it('should render', () => {
		const wrapper = shallow(<SkeletonLoading/>);

		expect(wrapper.exists()).toBeTruthy();
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
