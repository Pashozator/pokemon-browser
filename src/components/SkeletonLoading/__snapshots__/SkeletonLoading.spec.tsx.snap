// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonLoading component should render 1`] = `
<div
  className="skeleton-loading"
/>
`;
